typescript/infra/scripts/agents/deploy-agents.ts

```diff
@@ -107,7 +110,25 @@ async function main() {
   // run the create-keys script first.
   const { agentConfig } = await getConfigsBasedOnArgs();
   await checkDockerTagsExist(agentConfig);
-  await createAgentKeysIfNotExists(agentConfig);
+  const agentKeyToBeCreated = await isAgentKeyToBeCreated(agentConfig);
+
+  if (agentKeyToBeCreated) {
+    const shouldContinue = await confirm({
+      message: chalk.yellow.bold(
+        `Warning: New agent key will be created. Are you sure you want to continue?`,
```

Comment:
Could we surface what keys will be created? Otherwise I could see ppl not having enough info to know if this is expected or not. We could use the `key.identifier` string for each key